### DISCUSSION TITLE
How to launch SQLPage in a particular folder on Linux Mint at the booting ?

### DISCUSSION THREAD

**Discussion by ThierryM** [sqlpage/SQLPage#603] [Q&A] [answered]
Hi,
I want to launch automatically SQLPage which is on that folder /home/user/folder1 on Linux Mint 21.3.
So I use this command in Startup Applications:
`/home/user/folder1/sqlpage.bin`
But the default folder where SQLPage works is /home/user/sqlpage (this sqlpage folder is created at the boot time) and I have the classical page showing that SQLPage is working.
So how can I obtain the same behavior when I launch manually '/home/user/folder1/sqlpage.bin that works fine ?
Thanks for your help, regards.

**Comment by lovasoa** [comment, marked as answer]
Hi,

The problem with using "Startup Applications" in Gnome for launching applications like SQLPage is that it doesn't provide the same level of control as a dedicated system service. "Startup Applications" is more suitable for graphical applications or user-level processes, but for a service like SQLPage, it's better to define a [`systemd`](https://en.wikipedia.org/wiki/Systemd) service. This ensures that the application launches with the correct environment and configurations.

Here’s a guide on how to set up SQLPage to launch automatically using `systemd`:

### Steps to Create a Systemd Service for SQLPage

1. **Create a systemd service file**:
   
   You'll need to create a service file in `/etc/systemd/system/`. This file will define how and when SQLPage runs.

   Open a terminal and create the service file:
   ```bash
   sudo nano /etc/systemd/system/sqlpage.service
   ```

2. **Define the service file**:

   Inside the file, add the following content: https://github.com/lovasoa/SQLpage/blob/main/sqlpage.service

Important parameters:

   - **ExecStart**: This is the command that runs SQLPage. It points to the binary you mentioned (I recommend placing it in `/usr/local/bin/sqlpage.bin`).
   - **WorkingDirectory**: This sets the working directory. By default, if WEB_ROOT is not set, sqlpage serves files from this directory.
   - **Restart=always**: Ensures the service restarts if it crashes.
   - **User** and **Group**: These specify which user will run the service. Replace with your actual username.
   - **Environment**: Sets the [environment variables](https://github.com/lovasoa/SQLpage/blob/main/configuration.md).

3. **Reload systemd**:
   
   After creating the service file, you need to reload the systemd manager configuration so it recognizes the new service:
   ```bash
   sudo systemctl daemon-reload
   ```

4. **Enable the service to start at boot**:
   
   Enable the service so it starts automatically during boot:
   ```bash
   sudo systemctl enable sqlpage.service
   ```

5. **Start the service**:
   
   Start the service immediately to test if it works correctly:
   ```bash
   sudo systemctl start sqlpage.service
   ```

6. **Check the service status**:
   
   You can check if the service is running properly with:
   ```bash
   sudo systemctl status sqlpage.service
   ```

This method ensures SQLPage starts at boot and works in the specified directory. It provides more control over the execution, handling of environment variables, and the ability to restart the service if it fails.

If you want to stop the service at any time:
```bash
sudo systemctl stop sqlpage.service
```

This is a cleaner and more reliable approach than using "Startup Applications" for services like SQLPage that need to run in a specific context.

**Reply by lovasoa** [reply]
Once you have it running, to check the logs for your SQLPage `systemd` service, you can use the `journalctl` command. 

### 1. **Check recent logs for the SQLPage service**:

```bash
sudo journalctl -u sqlpage.service
```

This will show logs related to your SQLPage service in chronological order, starting from the oldest entry.

**Comment by ThierryM** [comment]
Thanks a lot Ophir for theses detailled explanations and the time you took to answer me !
I'll try as soon as possible.
Regards

**Comment by ThierryM** [comment]
Hi,
I've tested your method and it works perfectly. So I've rewrited in french here : https://lofurol.fr/joomla/logiciels-libres/104-bases-de-donnees/356-sqlpage-commandes-utiles-astuces
Thanks a lot again !
Regards
